Shopify best practices (#624)
rudderlabs/rudderstack-docs: docs/stream-sources/shopify.mdx

}
```

## Best practices

### Provide login functionality

> Shopify provides store owners with the capability to either make user login compulsory or allow guest checkout in their stores. However, it is a best practice to prompt  users to login on your platform. That ways, RudderStack fetches the user details to provide you with meaningful event data corresponding to the user's journey on your platform.

Reviewer: ```suggestion
Shopify provides the store owners the capability to either make user login compulsory or allow guest checkout in their stores. However, it is a best practice to prompt users to log in onto your platform. That way, RudderStack can fetch the user details and provide you with meaningful event data corresponding to the user's journey on your platform.
```

> ### Support for client side calls

Reviewer: ```suggestion
### Support for client-side calls
```

It is recommended **NOT** to make any drastic changes to your store structure in order to preserve data integrity. If done, RudderStack may lead to incorrect or missing calls owing to the disparity between URL and triggered event. 

> Refer to the following additional details for e-commerce events:

Reviewer: ```suggestion
Note to the following additional details for e-commerce events tracked by RudderStack:
```

You need to have the app installed and configured properly in each of the stores.

### What is the difference between `browser_ip` and `requestIP`?

> The `requestIP` is the IP address of Shopify backend server which pushes data to RudderStack while the `browser_ip` is the IP address of user's browser.

Reviewer: ```suggestion
The `requestIP` is the IP address of the Shopify backend server which pushes data to RudderStack, while `browser_ip` is the IP address of the user.
```

### Is it possible to collect additional/custom fields from Shopify?

Yes, it is possible to collect additional/custom fields from Shopify by using `identify` and `track` calls.

> Note that the `analytics` object of JS SDK (for example, `rudderanalytics.track()` or `rudderanalytics.identify()`) is already loaded on the `window` object of the store where the app is installed. Hence, the `analytics` object should not be loaded separately to collect additional/custom fields from Shopify.

Reviewer: ```suggestion
Note that the `rudderanalytics` object of JS SDK (for example, `rudderanalytics.track()` or `rudderanalytics.identify()`) is already loaded on the `window` object of the store where the app is installed. Hence, the `rudderanalytics` object should not be loaded separately to collect additional/custom fields from Shopify.
```

Reviewer: Confirmed with @yashasvibajpai re: this change. cc @utsabc @ishachopra0401

> ### How does RudderStack set Anonymous ID and can it persist?

Reviewer: ```suggestion
### How does RudderStack set the `anonymousId`? Does RudderStack persist it for future calls?
```

> RudderStack fetches Shopify's `customerId` and uses it as `anonymousId` to allow identity stitching in downstream destinations.

Reviewer: ```suggestion
RudderStack fetches Shopify's `customerId` and uses it as `anonymousId` to allow identity stitching in the downstream destinations.
```

> No, the `anonymousId` does not persist and keeps on changing as the Shopify servers push data to RudderStack over server to server API's.

Reviewer: ```suggestion
No, RudderStack does not persist the `anonymousId` - it keeps on changing as the Shopify servers push data to RudderStack over the server-to-server APIs.
```

> ### How can I differentiate between data collected from Shopify Cloud (Webhooks) and Shopify Device (Web Device Mode)?

Reviewer: ```suggestion
### How can I differentiate between the data collected from Shopify Cloud (Webhooks) and Shopify Device (web device mode)?
```

You can differentiate between the data collected in cloud and device mode by looking at the signature in payload:

> - In cloud mode, RudderStack subscribes to Shopify webhooks to consume data from Shopify. Data collected from Shopify web has the following signature in the payload:

Reviewer: @ishachopra0401  In the last sentence, should we change "Shopify web" to "Shopify Cloud"? To be consistent with the term used in the FAQ and in the payload below.
